erts/emulator/beam/erl_db_util.c

```diff
@@ -2804,6 +2805,72 @@ Eterm db_prog_match(Process *c_p,
                }
             }
             break;
+        case matchCurrentStacktrace: {
+            Uint sz;
+            Uint heap_size;
+            Eterm *hp;
+            Eterm mfa;
+            struct StackTrace *s;
+            int depth;
+            FunctionInfo* stk;
+            FunctionInfo* stkp;
+
+            ASSERT(c_p == self);
+
+            depth = unsigned_val(esp[-1]);
+
+            if (depth > erts_backtrace_depth) {
+                depth = erts_backtrace_depth;
+            }
+
+            sz = offsetof(struct StackTrace, trace) + sizeof(ErtsCodePtr) * depth;
+            s = (struct StackTrace *) erts_alloc(ERTS_ALC_T_TMP, sz);
+            s->depth = 0;
+            s->pc = NULL;
+
+            erts_save_stacktrace(c_p, s, depth);
+
+            depth = s->depth;
+            stk = stkp = (FunctionInfo *) erts_alloc(ERTS_ALC_T_TMP,
+					     depth*sizeof(FunctionInfo));
```

Comment:
Nitpick: please don't mix tabs and spaces, new code should be formatted using only spaces.